operator/webhooks/pod_mutator/pod_mutating_webhook.go

```diff
@@ -349,56 +347,56 @@ func (a *PodMutatingWebhook) handleWorkload(ctx context.Context, logger logr.Log
 }
 
 //nolint:dupl
-func (a *PodMutatingWebhook) handleApp(ctx context.Context, logger logr.Logger, pod *corev1.Pod, namespace string) error {
+func (a *PodMutatingWebhook) handleApp(ctx context.Context, logger logr.Logger, pod *corev1.Pod, namespace string, isAppAnnotationPresent bool) error {
 
 	ctx, span := a.Tracer.Start(ctx, "create_app", trace.WithSpanKind(trace.SpanKindProducer))
 	defer span.End()
 
-	newApp := a.generateApp(ctx, pod, namespace)
+	newAppCreationRequest := a.generateAppCreationRequest(ctx, pod, namespace, isAppAnnotationPresent)
 
-	newApp.SetSpanAttributes(span)
+	newAppCreationRequest.SetSpanAttributes(span)
 
-	logger.Info("Searching for app")
+	logger.Info("Searching for AppCreationRequest", "appCreationRequest", newAppCreationRequest.Name)
 
-	app := &klcv1alpha3.KeptnApp{}
-	err := a.Client.Get(ctx, types.NamespacedName{Namespace: namespace, Name: newApp.Name}, app)
+	appCreationRequest := &klcv1alpha3.KeptnAppCreationRequest{}
+	err := a.Client.Get(ctx, types.NamespacedName{Namespace: namespace, Name: newAppCreationRequest.Name}, appCreationRequest)
 	if errors.IsNotFound(err) {
-		logger.Info("Creating app", "app", app.Name)
-		app = newApp
-		err = a.Client.Create(ctx, app)
+		logger.Info("Creating app creation request", "appCreationRequest", appCreationRequest.Name)
+		appCreationRequest = newAppCreationRequest
+		err = a.Client.Create(ctx, appCreationRequest)
 		if err != nil {
 			logger.Error(err, "Could not create App")
-			controllercommon.RecordEvent(a.Recorder, apicommon.PhaseCreateApp, "Warning", app, "AppNotCreated", "could not create KeptnApp", app.Spec.Version)
+			controllercommon.RecordEvent(a.Recorder, apicommon.PhaseCreateApp, "Warning", appCreationRequest, "AppCreationRequestNotCreated", "could not create KeptnAppCreationRequest", appCreationRequest.Spec.AppName)
 			span.SetStatus(codes.Error, err.Error())
 			return err
 		}
 
-		controllercommon.RecordEvent(a.Recorder, apicommon.PhaseCreateApp, "Normal", app, "AppCreated", "created KeptnApp", app.Spec.Version)
+		controllercommon.RecordEvent(a.Recorder, apicommon.PhaseCreateApp, "Normal", appCreationRequest, "AppCreationRequestCreated", "created KeptnAppCreationRequest", appCreationRequest.Spec.AppName)
 		return nil
 	}
 
 	if err != nil {
 		span.SetStatus(codes.Error, err.Error())
-		return fmt.Errorf("could not fetch App"+": %+v", err)
+		return fmt.Errorf("could not fetch AppCreationRequest"+": %+v", err)
 	}
 
-	if reflect.DeepEqual(app.Spec, newApp.Spec) {
+	if reflect.DeepEqual(appCreationRequest.Spec, newAppCreationRequest.Spec) {
```

Comment:
Is the DeepEqual and the whole  update part needed here actually? The only information we have in the spec is the appName and this teoretically cannot change, that would lead to a new appCreationRequest. Or am I missing some info here ?

Reply:
good catch, yes that update logic is not needed anymore